unstable/wlr-output-power-management-unstable-v1.xml

```diff
@@ -0,0 +1,128 @@
+    interface version number is reset.
+  </description>
+
+  <interface name="zwlr_output_power_management_manager_v1" version="1">
+    <description summary="manager to create per-output power management">
+      This interface is a manager that allows creating per-output power
+      management mode controls.
+    </description>
+
+    <request name="get_output_power_management">
+      <description summary="get a power management for an output">
+        Create a power management mode control that can be used to
+        adjust the power management mode for a given output.
+      </description>
+      <arg name="id" type="new_id" interface="zwlr_output_power_management_v1"/>
+      <arg name="output" type="object" interface="wl_output"/>
+    </request>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        All objects created by the manager will still remain valid, until their
+        appropriate destroy request has been called.
+      </description>
+    </request>
+
+  </interface>
+  <interface name="zwlr_output_power_management_v1" version="1">
+    <description summary="adjust power save mode for an output">
+      This object is a manager which offers requests to set the power
+      save mode of outputs.
+    </description>
+
+    <enum name="mode">
+      <entry name="off" value="0"
+             summary="Output is turned off."/>
+      <entry name="on" value="1"
+             summary="Output is turned on, no power saving"/>
+    </enum>
+
+    <request name="set_mode">
+      <description summary="Set an outputs power save mode">
+        Set an output's power save mode to the given mode. The mode change
+        is effective immediately. If the output does not support the given
+        mode a failed event is sent.
+      </description>
+      <arg name="mode" type="uint" enum="mode" summary="the power save mode to set"/>
+    </request>
+
+    <event name="mode">
+      <description summary="Report a power save mode change">
+        Report the power save mode change of an output.
+
+        The mode event is sent after an output changed its power save mode. The reason
+        can be a client using set_mode or the compositor deciding to change an output's
+        mode.
+      </description>
+      <arg name="mode" type="uint" enum="mode"
+           summary="the output's new power save mode"/>
+    </event>
+
+    <event name="failed">
+      <description summary="object no longer valid">
+        This event indicates that the output power management control
+        is no longer valid. This can happen for a number of reasons,
+        including:
+        - The output doesn't support power management
+        - Another client already has exclusive power management control
+          for this output
+
+        Upon receiving this event, the client should destroy this object.
+      </description>
+    </event>
+
+    <event name="invalid_mode">
```

Comment:
This is racy -- if you issue two `set_mode` requests, then get an `invalid_mode` event, you don't know which mode isn't supported.

But I wonder if we need this event at all. Right now there's only on/off so the compositor can immediately send `failed` if the output doesn't support power management. It would be easy to add a new `supported_mode` event in the future to advertise a list of supported modes if we need it.

Reply:
o.k. let's drop that.